### PR TITLE
feat: Implement custom DatePicker and TimePicker for 2-way data binding

### LINKED ISSUE #603: bug: Passing ObservableField as parameter warning
**Is it a ?**

- [x] Bug Report

- [ ] Feature Request
    
- [ ] Chore
    

**Actual Behaviour**

Presently ObservableField is being passed as a parameter in DateBindings which create the following warning( refer the logcat ).

**Expected Behaviour**
Since the present method is deprecated, it should be changed.

**Steps to reproduce it**

This is a compile time warning.

**LogCat for the issue**

`Warning:Use of ObservableField and primitive cousins directly as method parameters is deprecated and support will be removed soon. Use the contents as parameters instead in method org.fossasia.openevent.app.common.app.binding.DateBindings.bindDate`

**Android version and Phone Model**

Moto G4 Plus    

**Would you like to work on the issue?**

Yes

### PR DESCRIPTION
**Pull request by srv-twry**
Fixes #603 

Changes: 

- Implemented custom DatePicker and TimePicker with 2-way data binding support.
- Delete DataBindings class since it is no longer required.
- Implement ```setListener``` and ```InverseBindingAdapters``` for both the custom views.

Screenshots for the change: 
videotogif_2018 05 11_02 21 34

**Update**
Changes: 
- Remove Observable fields and their related classes

### CODE REVIEW

**Comment by srv-twry** [comment]
@iamareebjamal Please review and suggest improvements. I believe we can extract out the common code between the two views into a common super class.

**Comment by iamareebjamal** [comment]
Yes, somewhat correct but can be improved. More important thing is that can we now remove the Observable String and such from POJOs, because that was causing warnings earlier

**Comment by srv-twry** [comment]
It works after converting the ObservableString to String in Ticket model. 
videotogif_2018 05 11_10 11 14


Why do we need to remove the ObservableString and such from all the POJOs. The deprecation warning  is due to the use of Observable fields directly in the ```BindingAdapter```. We have successfully removed it's usage.

**Comment by iamareebjamal** [comment]
Because we've been working on open event java library. It can't contain ObservableString in models, only pure Java types

**Comment by iamareebjamal** [comment]
But first check that the reactivity is working. Like the checking in attendee and everything. If it is not working, we can't remove these types yet. So first check those

**Comment by iamareebjamal** [comment]
Awesome job :)

**Comment by iamareebjamal** [comment]
I'm amazed at how the reactivity works even after removing the observable parts of the boolean and strings, is this all by the inverse adapters?

**Comment by srv-twry** [comment]
The binding and inverse binding adapters are only for the String. [This](https://medium.com/@douglas.iacovelli/custom-two-way-databinding-made-easy-f8b17a4507d2) medium blog explains how it all works.

As far as the boolean is concerned. I am not sure, we even needed them at the first place. I haven't written any adapter for boolean.

**Comment by srv-twry** [comment]
**Initially John wick checked in**

screenshot_2018-05-13-05-34-23-09

**Clicked on check out button, locally synced and job scheduled**
screenshot_2018-05-13-05-34-33-44

**checked out and synced from the server**
screenshot_2018-05-13-05-36-36-16


**Logs**
```
05-13 05:34:26.967 19309-19817/org.fossasia.eventyay D/JobProxy21: Schedule one-off jobInfo success, request{id=8, tag=attendee_check_in, transient=false}, start 00:00:00, end 00:00:05 (from now), reschedule count 0

05-13 05:36:28.445 19309-19401/org.fossasia.eventyay D/OkHttp: --> PATCH https://open-event-api-dev.herokuapp.com/v1/attendees/33?include=ticket,event,order&fields[event]=id&fields[ticket]=id
    Content-Type: application/vnd.api+json
    Content-Length: 556
    {"data":{"type":"attendee","id":"33","attributes":{"address":"address","attendee-delegate":{"header":"J","headerId":74,"identifier":33,"layoutRes":2131427360,"type":0,"enabled":true,"selectable":true,"selected":false},"checkin-times":"2018-05-13T05:36:28.314+05:30","city":"example","country":"IN","email":"payer.rich@example.com","firstname":"John","is-checked-in":false,"lastname":"Wick","state":"example","header":"J","header-id":74,"identifier":33,"layout-res":2131427360,"type":0,"checked-in":false,"enabled":true,"selectable":true,"selected":false}}}
    --> END PATCH (556-byte body)
05-13 05:36:29.435 20998-20998/? D/FloatWindowService: currentPackage=org.fossasia.eventyay, isHome=false
    currentPackage=org.fossasia.eventyay, isHome=false
05-13 05:36:29.445 20998-20998/? D/FloatWindowService: currentPackage=org.fossasia.eventyay, isHome=false
05-13 05:36:30.435 20998-20998/? D/FloatWindowService: currentPackage=org.fossasia.eventyay, isHome=false
    currentPackage=org.fossasia.eventyay, isHome=false
    currentPackage=org.fossasia.eventyay, isHome=false
05-13 05:36:30.675 19309-19401/org.fossasia.eventyay D/OkHttp: <-- 200 OK https://open-event-api-dev.herokuapp.com/v1/attendees/33?include=ticket,event,order&fields[event]=id&fields[ticket]=id (2228ms)
    Connection: keep-alive
    Server: gunicorn/19.7.1
    Date: Sun, 13 May 2018 00:06:30 GMT
    Content-Type: application/vnd.api+json
    Content-Length: 1425
    Etag: 4b622df792e2a93288b98ef1dff85a4c8b50d1dc
    Access-Control-Allow-Origin: *
    Via: 1.1 vegur
05-13 05:36:30.685 19309-19401/org.fossasia.eventyay D/OkHttp: {"included": [{"id": "5", "links": {"self": "/v1/events/5"}, "type": "event"}, {"id": "6", "links": {"self": "/v1/tickets/6"}, "type": "ticket"}], "links": {"self": "/v1/attendees/33"}, "data": {"relationships": {"event": {"links": {"related": "/v1/attendees/33/event", "self": "/v1/attendees/33/relationships/event"}, "data": {"id": "5", "type": "event"}}, "ticket": {"links": {"related": "/v1/attendees/33/ticket", "self": "/v1/attendees/33/relationships/ticket"}, "data": {"id": "6", "type": "ticket"}}, "user": {"links": {"related": "/v1/attendees/33/user", "self": "/v1/attendees/33/relationships/user"}}, "order": {"links": {"related": "/v1/attendees/33/order", "self": "/v1/attendees/33/relationships/order"}, "data": null}}, "attributes": {"is-checked-in": false, "blog": null, "firstname": "John", "facebook": null, "twitter": null, "phone": null, "company": null, "city": "example", "state": "example", "shipping-address": null, "ticket-id": "6", "gender": null, "birth-date": null, "country": "IN", "tax-business-info": null, "work-address": null, "email": "payer.rich@example.com", "github": null, "job-title": null, "billing-address": null, "address": "address", "checkin-times": "2018-05-13T05:36:28.314+05:30", "lastname": "Wick", "website": null, "pdf-url": null, "home-address": null, "work-phone": null}, "links": {"self": "/v1/attendees/33"}, "type": "attendee", "id": "33"}, "jsonapi": {"version": "1.0"}}
    <-- END HTTP (1425-byte body)

```

**Comment by iamareebjamal** [comment]
The booleans were for progress bar AFAIK

**Comment by mishuvs** [comment]
@srv-twry I created an attendee and tried checking in and checking out - It works for me too!

**Comment by iamareebjamal** [comment]
Alright, great job :smile: We can merge this after some semantic changes. My exams are going on so that's why there is a delay in reviews

**Comment by srv-twry** [comment]
@Masquerade0097 
Yes, you're right. It means we can't remove the ObservableBoolean yet.

**Comment by srv-twry** [comment]
Reverted back to ObservableBoolean in Faq.

**Comment by iamareebjamal** [comment]
Esther than observable boolean in FAQ. Use a map of observable boolean in presenter and use that in the data binding. Solves two issues

File: app/src/main/java/org/fossasia/openevent/app/core/faq/list/FaqListPresenter.java [hunk 1]
```diff
@@ -111,4 +117,15 @@ public void resetToDefaultState() {
         unselectFaq(previousFaq);
         getView().resetToolbar();
     }
+
+    public static ObservableBoolean getFaqSelected(Faq faq) {
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/app/core/faq/list/FaqListPresenter.java]: Why static?

**Comment by srv-twry** [comment]
Ready for review.

**Comment by iamareebjamal** [comment]
@srv-twry Please create an issue for polishing the implementation of custom views. As this PR was needed and was getting huge, we should iterate on a new issue

Nice work!

**Pull request merged**